src/backend/distributed/commands/utility_hook.c

```diff
@@ -237,8 +250,26 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		return;
 	}
 
-	ProcessUtilityInternal(pstmt, queryString, context,
-						   params, queryEnv, dest, completionTag);
+	UtilityHookLevel++;
+
+	PG_TRY();
+	{
+		bool ddlExecutedOnCitusTable = ProcessUtilityInternal(pstmt, queryString, context,
+															  params, queryEnv, dest,
+															  completionTag);
+		if (UtilityHookLevel == 1)
```

Comment:
some ideas to improve the flow a bit:
- passing the `ddlExecutedOnCitusTable ` flag down is hard to follow 
- Makes the calls to `ResetConstraintDropped ` way deep down and too implicit/error prone
- We should reset in the pg_catch as well because the transaction would rollback and the flag might be set to true



Please see https://github.com/citusdata/citus/commit/e4a3d97002d2a93f0282542c119460969e323632 for suggestions

Reply:
I will address that too

Reply:
I guess I addressed them in 592b817 & 2107c7d